src/shared/Atlassian.Bitbucket.Tests/BitbucketRestApiRegistryTest.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using GitCredentialManager;
+using Moq;
+using Xunit;
+
+namespace Atlassian.Bitbucket.Tests
+{
+    public class BitbucketRestApiRegistryTest
+    {
+        private Mock<ICommandContext> context = new Mock<ICommandContext>(MockBehavior.Strict);
+        private Mock<ISettings> settings = new Mock<ISettings>(MockBehavior.Strict);
+
+        [Fact]
+        public void BitbucketRestApiRegistry_Get_ReturnsCloudApi_ForBitbucketOrg()
+        {
+            // Given
+            settings.Setup(s => s.RemoteUri).Returns(new System.Uri("https://bitbucket.org"));
+            context.Setup(c => c.Settings).Returns(settings.Object);
+
+            var input = new InputArguments(new Dictionary<string, string>
+            {
+                ["protocol"] = "https",
```

Comment:
@mjcheetham The protocol is still needed because there is eventually a cal through to

        ```
public Uri GetRemoteUri(bool includeUser = false)
        {
            if (Protocol is null || Host is null)
            {
                return null;
            }
          ...
          }
```

Reply:
> @mjcheetham The protocol is still needed because there is eventually a call through to this method and fail if there is no protocol
> 
> 
> 
> ```
> 
>         public Uri GetRemoteUri(bool includeUser = false)
> 
>         {
> 
>             if (Protocol is null || Host is null)
> 
>             {
> 
>                 return null;
> 
>             }
> 
>           ...
> 
>           }
> 
> ```